ftd/src/executor/styles.rs

```diff
@@ -749,11 +816,153 @@ impl Resizing {
+                    line_number,
+                })?;
+            ftd::executor::Value::new(
+                ftd::executor::BackgroundRepeat::from_optional_value(
+                    value.clone(),
+                    doc,
+                    line_number,
+                )?,
+                Some(line_number),
+                vec![value.into_property(ftd::interpreter2::PropertySource::header("repeat"))],
+            )
+        };
+
+        let size = {
+            let value = values.get("size").ok_or(ftd::executor::Error::ParseError {
+                message: "`size` field in ftd.background-image not found".to_string(),
+                doc_id: doc.name.to_string(),
+                line_number,
+            })?;
+            ftd::executor::Value::new(
+                ftd::executor::BackgroundSize::from_optional_value(
+                    value.clone(),
+                    doc,
+                    line_number,
+                )?,
+                Some(line_number),
+                vec![value.into_property(ftd::interpreter2::PropertySource::header("size"))],
+            )
+        };
+
+        let position = {
+            let value = values
+                .get("position")
+                .ok_or(ftd::executor::Error::ParseError {
+                    message: "`position` field in ftd.background-image not found".to_string(),
+                    doc_id: doc.name.to_string(),
+                    line_number,
+                })?;
+            ftd::executor::Value::new(
+                ftd::executor::BackgroundPosition::from_optional_value(
+                    value.clone(),
+                    doc,
+                    line_number,
+                )?,
+                Some(line_number),
+                vec![value.into_property(ftd::interpreter2::PropertySource::header("position"))],
+            )
+        };
+
+        Ok(BackgroundImage {
+            src,
+            repeat,
+            size,
+            position,
+        })
+    }
+
+    pub fn to_image_src_css_string(&self) -> String {
+        format!("url({})", self.src.value.light.value)
+    }
+
+    pub fn to_repeat_css_string(&self) -> String {
+        match self.repeat.value.as_ref() {
+            Some(s) => s.to_css_string(),
+            None => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+        }
+    }
+
+    pub fn to_size_css_string(&self) -> String {
+        match self.size.value.as_ref() {
+            Some(s) => s.to_css_string(),
+            None => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+        }
+    }
+
+    pub fn to_position_css_string(&self) -> String {
+        match self.position.value.as_ref() {
+            Some(s) => s.to_css_string(),
+            None => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+        }
+    }
+}
 
 #[derive(serde::Deserialize, Debug, PartialEq, Clone, serde::Serialize)]
 pub enum Background {
     Solid(Color),
-    Image(ftd::executor::ImageSrc),
+    Image(BackgroundImage),
```

Comment:
Give full reference `BackgroundImage` to `ftd::executor::BackgroundImage` and `Color` to `ftd::executor::Color`